### ISSUE TITLE
Ruff extension for Visual Studio Code not show error E501 (line-too-long)

### ISSUE THREAD

**Issue by alejandro-kid** [astral-sh/ruff#12096]
I am using Visual Studio Code Insider, version 1.85, in Fedora Linux version 39.
Don't matter what config I set, never shows when the line exceeds de configured limit. The other features in RUFF work well.

**Comment by charliermarsh** [comment]
We actually don't enable `E501` by default. Are you enabling it in your configuration file?

**Comment by Denis-Alexeev** [comment]
@charliermarsh I have found 1 case when ruff does not raise E501. Pls tell me if I should create new issue for that.
This is steps to reproduce.

I have 2 modules: `other.py` and `test_new.py`

`other.py`
```python
def very_very_very_long_function_name(attr: int):
    pass

```

`test_new.py`
```python
"""docstring."""

import logging

import other

logging.info("first long")
other.very_very_very_long_function_name(9)
logging.info("second long")

```

`ruff.toml`
```toml
target-version = "py312"
line-length = 17

[lint]
select = ["ALL"]

[lint.pydocstyle]
convention = "google"
```

```bash
ruff --version                                                       
ruff 0.5.1

python --version   
Python 3.12.2
```

`command`
```bash
ruff check --config ./ruff.toml --output-format concise ./test_new.py
```

`output`
```bash
test_new.py:7:18: E501 Line too long (26 > 17)
test_new.py:9:18: E501 Line too long (27 > 17)
Found 2 errors.
```

`Expected behavior`
3 violations of E501

`Actual behavior`
2 violations of E501

**Comment by MichaReiser** [comment]
@Denis-Alexeev The problem here is that the line 8 contains no whitespace ([see the rules documentation](https://docs.astral.sh/ruff/rules/line-too-long/#why-is-this-bad)). 

I think we could do better here now that is kind of cheap to get the tokens at a specific offset by only applying the exception rules if the range falls into a string (wdyt @charliermarsh )?

**Comment by Denis-Alexeev** [comment]
@MichaReiser Oh, I guess I've misread the first exception. I always thought, that it is about exactly one word. Now I see that the word "word" has quotation marks 😄
